examples/grpc/src/main/java/io/opentelemetry/example/HelloWorldClient.java

```diff
@@ -0,0 +1,166 @@
+
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class HelloWorldClient {
+  private static final Logger logger = Logger.getLogger(HelloWorldClient.class.getName());
+  private final ManagedChannel channel;
+  private final String serverHostname;
+  private final Integer serverPort;
+  private final GreeterGrpc.GreeterBlockingStub blockingStub;
+
+  // OTel API
+  Tracer tracer =
+      OpenTelemetry.getTracerFactory().get("io.opentelemetry.example.HelloWorldClient");;
+  // Export traces as log
+  LoggingExporter exporter = new LoggingExporter();
+  // Share context via text headers
+  HttpTextFormat<SpanContext> textFormat = tracer.getHttpTextFormat();
+  // Inject context into the gRPC request metadata
+  HttpTextFormat.Setter<Metadata> setter =
+      new HttpTextFormat.Setter<Metadata>() {
+        @Override
+        public void put(Metadata carrier, String key, String value) {
+          carrier.put(Metadata.Key.of(key, Metadata.ASCII_STRING_MARSHALLER), value);
+        }
+      };
+
+  /** Construct client connecting to HelloWorld server at {@code host:port}. */
+  public HelloWorldClient(String host, int port) {
+    this.serverHostname = host;
+    this.serverPort = port;
+    this.channel =
+        ManagedChannelBuilder.forAddress(host, port)
+            // Channels are secure by default (via SSL/TLS). For the example we disable TLS to avoid
+            // needing certificates.
+            .usePlaintext()
+            // Intercept the request to tag the span context
+            .intercept(new OpenTelemetryClientInterceptor())
+            .build();
+    blockingStub = GreeterGrpc.newBlockingStub(channel);
+    // Initialize the OTel tracer
+    initTracer();
+  }
+
+  private void initTracer() {
+    // Use the OpenTelemetry SDK
+    TracerSdkFactory tracerFactory = OpenTelemetrySdk.getTracerFactory();
+    // Set to process the spans by the log exporter
+    tracerFactory.addSpanProcessor(SimpleSpansProcessor.newBuilder(exporter).build());
+  }
+
+  public void shutdown() throws InterruptedException {
+    channel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
+  }
+
+  /** Say hello to server. */
+  public void greet(String name) {
+    logger.info("Will try to greet " + name + " ...");
+
+    // Start a span
+    Span span =
+        tracer.spanBuilder("helloworld.Greeter/SayHello").setSpanKind(Span.Kind.CLIENT).startSpan();
+    // TODO provide attributes to Span.Builder
+    span.setAttribute("component", "grpc");
+    span.setAttribute("rpc.service", "Greeter");
+    span.setAttribute("net.peer.ip", this.serverHostname);
+    span.setAttribute("net.peer.port", this.serverPort);
+
+    // Set the context with the current span
+    try (Scope scope = tracer.withSpan(span)) {
+      HelloRequest request = HelloRequest.newBuilder().setName(name).build();
+      try {
+        HelloReply response = blockingStub.sayHello(request);
```

Comment:
It might be useful to have a grpc streaming example as well. I assume it would share much of the same code, so maybe included in the same sample?

Reply:
Make sense to me. I will add it!